druid/system/druid_instance.lua

```diff
@@ -249,6 +277,10 @@ function DruidInstance.initialize(self, context, style)
 	self._input_blacklist = nil
 	self._input_whitelist = nil
 
+	self.set_pr = Event()
```

Comment:
Also all of these events are redundant and things it's solved can be done via already existing Druid API